Compile error c++ (GCC) 7.2.1 20170915 (not a constant expression)

With `c++ (GCC) 7.2.1 20170915` (i.e. Fedora 27 default GCC version), the following compiler error is generated:
```
In file included from <snip>/media_driver/agnostic/common/os/mos_context.cpp:28:0:
<snip>/media_driver/linux/common/os/mos_context_specific.h:63:67: error: value ‘18446744073709551615’ of type ‘void*’ is not a constant expression
     constexpr static void* MOS_LINUX_SHM_INVALID       = (void *)-1;
                                                                   ^
```

--- a/media_driver/linux/common/os/mos_context_specific.cpp
+++ b/media_driver/linux/common/os/mos_context_specific.cpp
@@ -134,7 +134,7 @@ MOS_STATUS OsContextSpecific::DestroySemaphore(unsigned int semid)
     return MOS_STATUS_SUCCESS;
 }
 
-MOS_STATUS OsContextSpecific::ConnectCreateShm(long key, uint32_t size, uint32_t *pShmid, void* *ppShm)
+MOS_STATUS OsContextSpecific::ConnectCreateShm(long key, uint32_t size, int32_t *pShmid, void* *ppShm)
 {
     struct shmid_ds buf;
     int32_t         shmid;
@@ -196,7 +196,7 @@ MOS_STATUS OsContextSpecific::DetachDestroyShm(int32_t shmid, void  *pShm)
     return MOS_STATUS_SUCCESS;
 }
 
-MOS_STATUS OsContextSpecific::ConnectCreateSemaphore(long key, uint32_t *pSemid)
+MOS_STATUS OsContextSpecific::ConnectCreateSemaphore(long key, int32_t *pSemid)
 {
     int32_t         semid;
     struct sembuf   sop;

--- a/media_driver/linux/common/os/mos_context_specific.h
+++ b/media_driver/linux/common/os/mos_context_specific.h
@@ -60,12 +60,12 @@ class OsContextSpecific : public OsContext
     //!
     //! \brief Initial share memory handle
     //!
-    constexpr static void* MOS_LINUX_SHM_INVALID       = (void *)-1;
+    constexpr static void* MOS_LINUX_SHM_INVALID       = (void *)nullptr;
 
     //!
     //! \brief Initial share memory ID
     //!
-    constexpr static uint32_t MOS_LINUX_IPC_INVALID_ID = -1;
+    constexpr static int32_t MOS_LINUX_IPC_INVALID_ID = -1;
 
     //!
     //! \brief maximum number to try to get a valid semaphore
@@ -131,13 +131,12 @@ public:
     //! \brief  Return the semaphore ID we use to protect the IPC creation process
     //! \return sem id
     //!
-    uint32_t GetSemId(){return m_semId;};
+    int32_t GetSemId(){return m_semId;};
 
     //!
-    //! \brief  Return the shm ID for the IPC
-    //! \return shm id
-    //!
-    uint32_t GetShmId(){return m_shmId;};
+    //! \brief  Return the shm ID for the gPC
+    //! \return shm gd
+    int32_t GetShmId(){return m_shmId;};
 
     //!
     //! \brief  Return the shm object for the IPC
@@ -178,7 +177,7 @@ private:
     //!         ptr to ptr for share memory
     //! \return MOS_SUCCESS in success case, MOS error status in fail cases
     //!
-    MOS_STATUS ConnectCreateShm(long key, uint32_t size, uint32_t * pShmid, void* *ppShm);
+    MOS_STATUS ConnectCreateShm(long key, uint32_t size, int32_t * pShmid, void* *ppShm);
 
     //!
     //! \brief  destory the share memory
@@ -198,7 +197,7 @@ private:
     //!         ptr to sem id created
     //! \return MOS_SUCCESS in success case, MOS error status in fail cases
     //!
-    MOS_STATUS ConnectCreateSemaphore(long key, uint32_t *pSemid);
+    MOS_STATUS ConnectCreateSemaphore(long key, int32_t *pSemid);
 
     //!
     //! \brief  create driver secure IPC
@@ -276,11 +275,11 @@ private:
     //!
     //! \brief  Semophore ID for secure IPC
     //!
-    uint32_t            m_semId = 0;
+    int32_t            m_semId = 0;
     //!
     //! \brief  Share memory ID for secure IPC
     //!
-    uint32_t            m_shmId = 0;
+    int32_t            m_shmId = 0;
     //!
     //! \brief  Share memory ptr for secure IPC
     //!
@@ -299,11 +298,11 @@ private:
     //!
     //! \brief  Semophore ID for ruling SSEU configration
     //!
-    uint32_t            m_sseuSemId = 0;
+    int32_t            m_sseuSemId = 0;
     //!
     //! \brief  Share memory ID for ruling SSEU configration
     //!
-    uint32_t            m_sseuShmId = 0;
+    int32_t            m_sseuShmId = 0;
     //!
     //! \brief  Share memory ptr to the ruling SSEU configration
     //!

